### PR TITLE
Support PyTorch1.13 and above

### PR DESCRIPTION
**Pull request by PenghuiCheng**
# What does this PR do?

This PR will supported PyTorch 1.13.0 and above version quantization. In PyTorch 1.13.0, there is a parameter: [example_input](https://github.com/pytorch/pytorch/blob/dc00bb51b8d370bf3891f0edb2c6e0c2914e329a/torch/ao/quantization/quantize_fx.py#L271) in prepare_fx function. So we should specify the dataloader to optimum for static quantization and quantization-aware training.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by echarlaix — COMMENTED** [review]
Thanks for this PR @PenghuiCheng 🔥

File: optimum/intel/neural_compressor/quantization.py [hunk 1]
```diff
@@ -110,6 +112,9 @@ def __init__(
             if self.train_func is None:
                 raise ValueError("train_func must be provided for quantization aware training.")
             self.quantization.q_func = self.train_func
+            if version.release >= Version("1.13.0").release:
+                assert self.calib_dataloader is not None, "calib_dataloader must be provided for PyTorch1.13 or above."
```
[the remark this answers is not in the document]
Comment (PenghuiCheng) [reply optimum/intel/neural_compressor/quantization.py]: Done

File: optimum/intel/neural_compressor/utils.py [hunk 2]
```diff
@@ -56,13 +59,28 @@ def _cfgs_to_fx_cfgs(op_cfgs: Dict, observer_type: str = "post_training_static_q
         fx_op_cfgs (`dict`):
             Dictionary of quantization configure that meets the requirements of torch.fx.
     """
-    fx_op_cfgs = dict()
-    op_tuple_cfg_list = []
+    version = get_torch_version()
```
[the remark this answers is not in the document]
Comment (PenghuiCheng) [reply optimum/intel/neural_compressor/utils.py]: Done

File: tests/neural_compressor/test_optimization.py [hunk 3]
```diff
@@ -410,6 +379,43 @@ def eval_func(model):
...
+            def eval_func(model):
+                trainer.model = model
+                metrics = trainer.evaluate()
+                return metrics["eval_accuracy"]
+
+            config_path = os.path.dirname(os.path.abspath(__file__))
+            q8_config = IncQuantizationConfig.from_pretrained(config_path)
+            q8_config.set_config("quantization.approach", IncQuantizationMode.AWARE_TRAINING.value)
+            q8_config.set_config("tuning.accuracy_criterion.relative", 0.2)
+            q8_config.set_config("model.framework", "pytorch_fx")
```
[the remark this answers is not in the document]
Comment (PenghuiCheng) [reply tests/neural_compressor/test_optimization.py]: Done

**Review by echarlaix — APPROVED** [review]
Thanks for this addition @PenghuiCheng !

**Pull request merged**